### PR TITLE
[Templates] Migrate templates into workers-sdk monorepo

### PR DESCRIPTION
**Pull request by lauragift21**
What this PR solves / how to test:
Moving template from its own repo into the Wrangler monorepo for more direct support, maintenance, and improvements. 

Follow up of https://github.com/cloudflare/workers-sdk/pull/2498

### CODE REVIEW

**Comment by penalosa** [comment]
@lauragift21 Would it be possible to move these experimental templates into an `experimental` subfolder of the `templates` folder?

**Comment by lauragift21** [comment]
> @lauragift21 Would it be possible to move these experimental templates into an `experimental` subfolder of the `templates` folder?

Sure,@penalosa I can do that. I will update the PR.

**Comment by JacobMGEvans** [comment]
It looks like a machine generated `.idea` folder `templates/experimental/worker-dart/.idea` looks like it is generated from intelliJ on OSX. It should be gitignored at the monorepo root if it isn't already.

**Review by JacobMGEvans — APPROVED** [review]
We should look into converting all the templates into module workers. There is also the `.idea` that needs to be removed from `templates/experimental/worker-dart/.idea` and added to a top level `.gitignore`

Converting templates to module workers should be a larger effort and done for all the templates.

**Comment by lrapoport-cf** [comment]
> > @lauragift21 Would it be possible to move these experimental templates into an `experimental` subfolder of the `templates` folder?
> 
> Sure,@penalosa I can do that. I will update the PR.

@lauragift21 @penalosa @JacobMGEvans should we also include the BinAST template in the experimental subfolder?

**Comment by lauragift21** [comment]
@lrapoport-cf  I think that template is stand-alone it's not part of the experimental templates.

**Comment by lauragift21** [comment]
> We should look into converting all the templates into module workers. There is also the `.idea` that needs to be removed from `templates/experimental/worker-dart/.idea` and added to a top level `.gitignore`
> 
> Converting templates to module workers should be a larger effort and done for all the templates.

@JacobMGEvans yes I agree they should all use module workers templates, I can take this on after this is done.

**Comment by penalosa** [comment]
I think BinAST should probably be in the experimental folder as well? The Binary AST proposal is still a proposal, and seems pretty slow-going right now (and not great in terms of browser support, unless I'm missing something). https://github.com/tc39/proposal-binary-ast/issues/79

**Comment by JacobMGEvans** [comment]
> > We should look into converting all the templates into module workers. There is also the `.idea` that needs to be removed from `templates/experimental/worker-dart/.idea` and added to a top level `.gitignore`
> > Converting templates to module workers should be a larger effort and done for all the templates.
> 
> @JacobMGEvans yes I agree they should all use module workers templates, I can take this on after this is done.

I created an issue to track this, I will assign you! Thanks for taking it on 🥳

**Comment by lrapoport-cf** [comment]
@lauragift21 thanks for doing this! to confirm, for each of the templates being migrated here, are these their originating repos? 

- `worker-dart`: https://github.com/cloudflare/dart-worker-hello-world
- `worker-fsharp`: https://github.com/fable-compiler/cfworker-hello-world
- `worker-kotlin`: https://github.com/cloudflare/kotlin-worker-hello-world
- `worker-perl`: https://github.com/cloudflare/perl-worker-hello-world
- `worker-php`: https://github.com/cloudflare/php-worker-hello-world
- `worker-python`: https://github.com/cloudflare/python-worker-hello-world
- `worker-reason`: https://github.com/cloudflare/reason-worker-hello-world
- `worker-scala-kv`: https://github.com/cloudflare/scala-worker-kv
- `worker-scala`: https://github.com/cloudflare/scala-worker-hello-world

my understanding is that original repos should be deprecated after this migration, is that correct?

**Comment by lauragift21** [comment]
> @lauragift21 thanks for doing this! to confirm, for each of the templates being migrated here, are these their originating repos?
> 
> * `worker-dart`: https://github.com/cloudflare/dart-worker-hello-world
> * `worker-fsharp`: https://github.com/fable-compiler/cfworker-hello-world
> * `worker-kotlin`: https://github.com/cloudflare/kotlin-worker-hello-world
> * `worker-perl`: https://github.com/cloudflare/perl-worker-hello-world
> * `worker-php`: https://github.com/cloudflare/php-worker-hello-world
> * `worker-python`: https://github.com/cloudflare/python-worker-hello-world
> * `worker-reason`: https://github.com/cloudflare/reason-worker-hello-world
> * `worker-scala-kv`: https://github.com/cloudflare/scala-worker-kv
> * `worker-scala`: https://github.com/cloudflare/scala-worker-hello-world
> 
> my understanding is that original repos should be deprecated after this migration, is that correct?

Yes you're correct @lrapoport-cf. The follow-up task is to deprecate or archive these repos.

File: .gitignore [hunk 1]
```diff
@@ -188,3 +188,6 @@ _routes.generated.json
 
 # Vendored npm dependencies
 !vendor/*.tgz
+
+# IntelliJ 
+.idea/
```
Comment (JacobMGEvans) [inline .gitignore:193]:
```suggestion
.idea/

```

File: templates/experimental/worker-dart/worker-binast/wrangler.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+name = "binast"
+main = "index.js"
+compatibility_date = "2023-02-22"
```
Comment (JacobMGEvans) [inline templates/experimental/worker-dart/worker-binast/wrangler.toml:3]:
```suggestion
compatibility_date = "2023-02-22"

```

File: templates/experimental/worker-fsharp/.gitignore [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+#ionide
+symbolCache.db
+
+
+# Wrangler/Workers
+/target
+/dist
+worker/
+tmp/
```
Comment (JacobMGEvans) [inline templates/experimental/worker-fsharp/.gitignore:17]:
```suggestion
tmp/

```

File: templates/experimental/worker-fsharp/src/FetchUtils.fs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module Fetch.Utils
+
+open Fetch
+open Fable.Core
+
+[<Emit("new Response($0, $1)")>]
+let newResponse (a:string) (b:ResponseInit) : Response = jsNative
```
Comment (JacobMGEvans) [inline templates/experimental/worker-fsharp/src/FetchUtils.fs:7]:
```suggestion
let newResponse (a:string) (b:ResponseInit) : Response = jsNative

```

File: templates/experimental/worker-kotlin/.gitignore [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+wasm-pack.log
+worker/
+node_modules/
+.cargo-ok
+/index.js
+/build/
+.idea
+.gradle
+.DS_Store
+kotlin-js-store
```
Comment (JacobMGEvans) [inline templates/experimental/worker-kotlin/.gitignore:15]:
```suggestion
kotlin-js-store

```

File: templates/experimental/worker-perl/wrangler.toml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+name = "worker-perl"
+main = "index.js"
+compatibility_date = "2023-02-22"
+node_compat = true
+
+[build]
+command = "npm run build"
```
Comment (JacobMGEvans) [inline templates/experimental/worker-perl/wrangler.toml:7]:
```suggestion
command = "npm run build"

```

File: templates/experimental/worker-kotlin/cargo-generate.toml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+[template]
+exclude = ["*.jar"]
```
Comment (JacobMGEvans) [inline templates/experimental/worker-kotlin/cargo-generate.toml:2]:
```suggestion
exclude = ["*.jar"]

```

File: templates/experimental/worker-kotlin/gradle.properties [hunk 8]
```diff
@@ -0,0 +1,2 @@
+kotlin.code.style=official
+kotlin.js.ir.output.granularity=whole-program
```
Comment (JacobMGEvans) [inline templates/experimental/worker-kotlin/gradle.properties:2]:
```suggestion
kotlin.js.ir.output.granularity=whole-program

```

File: templates/experimental/worker-php/index.php [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+addEventListener("fetch", function ($event) {
+    $event->respondWith(handleRequest($event->request));
+});
+
+function handleRequest($request) {
+    return new Response("PHP Worker hello world", [
+        "headers" => [ "content-type" => "text/plain" ]
+    ]);
+}
```
Comment (JacobMGEvans) [inline templates/experimental/worker-php/index.php:11]:
```suggestion
}

```

File: templates/experimental/worker-php/wrangler.toml [hunk 10]
```diff
@@ -0,0 +1,6 @@
+name = "worker-php"
+main = "index.js"
+compatibility_date = "2023-02-22"
+
+[build]
+command = "npm run build"
```
Comment (JacobMGEvans) [inline templates/experimental/worker-php/wrangler.toml:6]:
```suggestion
command = "npm run build"

```

File: templates/experimental/worker-scala/wrangler.toml [hunk 11]
```diff
@@ -0,0 +1,2 @@
+name = "worker-scala"
+main = "src/main/scala/Main.scala"
```
Comment (JacobMGEvans) [inline templates/experimental/worker-scala/wrangler.toml:2]:
```suggestion
main = "src/main/scala/Main.scala"

```

**Review by JacobMGEvans — APPROVED** [review]
A lot of new file types that our Prettier config isn't handling 😆 

re-approving pending these minor codebase formatting.

**Pull request merged**